### ISSUE TITLE
Make colors WCAG compliant by default

### ISSUE THREAD

**Issue by jouni** [vaadin/vaadin-lumo-styles#22]
The current success color does not pass the contrast test. [We indicate that in the docs](https://cdn.vaadin.com/vaadin-lumo-styles/1.0.0-beta3/demo/colors.html#success), but it’s still poor service and bad service for users.

**Comment by vlukashov** [comment]
The `primary` color is not compliant either. It has the contrast ratio of 4.27 which is lower than 4.5 required by [WCAG 2.0 Level AA](https://www.w3.org/WAI/WCAG21/quickref/#contrast-minimum).

```css
:host([theme~="primary"]) {
  color: var(--lumo-primary-contrast-color);
  background-color: var(--lumo-primary-color);
}
```

**Comment by anezthes** [comment]
Here's a quick fix:
![colors](https://user-images.githubusercontent.com/7125728/77895392-5fafef00-727f-11ea-8427-3ce05059b05a.png)

**Comment by jouni** [comment]
Some might consider this as a breaking change. I’m on the fence – on one hand, if you haven’t touched the defaults, I think you’d appreciate that we fix the contrast issue, but on the other hand maybe you’ve tweaked some other parts of your theme to fit nicely with the default colors.

It’s at least worth a minor version upgrade, but I’m hesitant to pump the major version for this.

@anezthes could you do the same for the dark palette?

**Comment by anezthes** [comment]
You're welcome!

I personally don't mind if it's a minor or major version as long as it's fixed at some point And just to be clear; I'm not saying we need to use these exact values, I just wanted to provide _one_ solution so that we hopefully fix it sooner rather than later.

I'll see if I can do the same for the dark palette.

**Comment by anezthes** [comment]
Dark color palette:
* --lumo-primary-color
hsl(214, 86%, 55%) → hsl(214, 85%, 50%)

* --lumo-success-color
hsl(145, 65%, 42%) → hsl(145, 65%, 32%)

* --lumo-error-color
hsl(3, 90%, 63%) → hsl(3, 90%, 48%)

* --lumo-error-text-color
hsl(3, 100%, 67%) → hsl(3, 100%, 74%)

![dark-colors](https://user-images.githubusercontent.com/7125728/78107510-45e8e600-73fe-11ea-827e-af9a65081ad3.png)

**Comment by anezthes** [comment]
Checked the **badges** as well.

The **light** theme fails to meet the contrast criteria on two occasions, even with the fixes provided above, and those are:
* `theme="badge"` (4.43)
* `theme="badge error"` (4.42)

Those can be fixed by turning down the luminosity by 1 on both `var(--lumo-primary-text-color)` and `var(--lumo-error-text-color)`. In other words:

* `--lumo-primary-text-color`
`hsl(214, 90%, 47%)` → `hsl(214, 90%, 46%)` (4.58)

* `--lumo-error-text-color`
`hsl(3, 92%, 45%)` → `hsl(3, 92%, 44%)` (4.6)

**Comment by jouni** [comment]
I wanted to add this as a note, that the reason why the default colors are what they are currently is because I was aiming for 3:1 contrast initially.  The success color is the only one that fails there.

I considered the medium (500) font weight for the secondary buttons and the semi bold (600) for the primary buttons, together with the default 16px font size, to satisfy WCAG AA requirements for “large or bold text”. The 4.5:1 contrast is required when “text is less than 18 point if not bold and less than 14 point if bold”.

But I think I failed to notice that the spec is actually talking about points instead of pixels for font sizes. “Less than 14 point if bold” equals “less than 18.5px if bold”, so 16px is not enough to satisfy that. And the small button variant is of course worse.

**Comment by rolfsmeds** [comment]
This could be fixed as part of a non-LTS release and backported to the next (post-14) LTS version, to avoid breaking changes.

**Comment by web-padawan** [comment]
Closing in favor of https://github.com/vaadin/vaadin-web-components/issues/119 which describes the same problem.
